### PR TITLE
Updated button role interactions and added NSA category blacklist

### PR DESCRIPTION
**Pull request by NathanealV**

### CODE REVIEW

File: src/main/java/com/ibdiscord/listeners/ButtonListener.java [hunk 1]
```diff
@@ -60,10 +60,11 @@ public void onButtonClick(@NotNull ButtonClickEvent event) {
                 toAdd.add(role);
             }
         }
-        guild.modifyMemberRoles(member, toAdd, new ArrayList<>()).queue(yes -> {
-            event.reply(generateMessage(toAdd.size(), 0))
-                .setEphemeral(true)
-                .queue();
+        event.deferReply(true).queue(deferred -> {
+                guild.modifyMemberRoles(member, toAdd, new ArrayList<>()).queue(yes -> {
+                    event.getHook().sendMessage(generateMessage(toAdd.size(), 0))
```
Comment (Arraying) [inline src/main/java/com/ibdiscord/listeners/ButtonListener.java:65]: Does this stop the loading... animation when it sends the message after a while? What happens if the `sendMessage` gets called like 10 seconds later?

File: bot-example.env [hunk 2]
```diff
@@ -44,6 +44,11 @@ SENSITIVE_ROLES=
...
-# A list of explicitly opt-blacklisted channel IDs .
+# A list of explicitly opt-blacklisted channel IDs.
 # Separated by ;
-OPT_BLACKLIST=
\ No newline at end of file
+OPT_BLACKLIST=
+
+# A list of category IDs in which NSA is disabled.
+# Separated by ;
+NSA_CATEGORIES=
```
Comment (Arraying) [inline bot-example.env:53]: Unintuitive name; please change to something like NSA_BLACKLIST (or to be more politically correct, NSA_DENYLIST).

**Review by Arraying — CHANGES_REQUESTED** [review]
Looks good (if it is tested), but please renamed the field.

File: bot-example.env (hunk 2 above)
Comment (NathanealV) [reply bot-example.env]: I've changed the name to NSA_DENYLIST.

File: src/main/java/com/ibdiscord/listeners/ButtonListener.java (hunk 1 above)
Comment (NathanealV) [reply src/main/java/com/ibdiscord/listeners/ButtonListener.java]: I've tested artificial delays (with Thread.sleep) and it works as intended (e.g., continues loading animation and then edits the message after action completes).

**Review by Arraying — APPROVED** [review]
Thanks for the changes.

**Pull request merged**